compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -570,8 +570,14 @@ impl<'ll, 'tcx> MiscMethods<'tcx> for CodegenCx<'ll, 'tcx> {
     }
 
     fn declare_c_main(&self, fn_type: Self::Type) -> Option<Self::Function> {
-        if self.get_declared_value("main").is_none() {
-            Some(self.declare_cfn("main", llvm::UnnamedAddr::Global, fn_type))
+        let entry_name = self.sess().target.entry_name.as_ref();
+        if self.get_declared_value(entry_name).is_none() {
+            Some(self.declare_entry_fn(
+                entry_name,
+                self.sess().target.entry_abi.into(),
```

Comment:
rustc_codegen_gcc and rustc_codegen_cranelift also need this change.

Reply:
Should I add that in a different PR or update this one?

Reply:
A separate PR is fine by me. But if you want to update this PR, I can review it too.

Reply:
Ok, So I'm not sure how to approach gcc. As is visible [here](https://github.com/rust-lang/rust/blob/357f66072959f494d5fa2e6a269c6100eed703c1/compiler/rustc_codegen_gcc/src/declare.rs#L68), the entry function signature and call convention is hard-coded there. So there is not much point in changing the entry name until the `TODOs` are fixed.

Reply:
For cg_clif the change would be in src/main_shim.rs. As for cg_gcc cc @antoyo.

Reply:
I believe the same change as you did for LLVM applies: it's just that the function type is hard-coded for now. You can keep it that way.